### PR TITLE
Prevent movedRemainingGasToChild flag from being reverted to false on contracts with multiple CALL op codes

### PR DESCRIPTION
**Pull request by illuque-iov**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Comment by illuque-iov** [comment]
pipeline:run

**Comment by illuque-iov** [comment]
pipeline:run

**Review by Vovchyk — CHANGES_REQUESTED** [review]

File: rskj-core/src/main/java/org/ethereum/vm/program/ProgramResult.java [hunk 1]
```diff
@@ -68,6 +68,10 @@ public class ProgramResult {
     private long gasUsedBeforeRefunds = 0; // this field it's useful to test if the deductedRefund value is less than the half of the gasUsed
 
     public void movedRemainingGasToChild(boolean moved) {
+        // once market as true, this should prevail
+        if (this.movedRemainingGasToChild) {
```
[the remark this answers is not in the document]
Comment (illuque-iov) [reply rskj-core/src/main/java/org/ethereum/vm/program/ProgramResult.java:72]: Yes, much better. I thought I was going to use to for testing but then I didn't. Will change it.

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java [hunk 2]
```diff
@@ -184,6 +184,9 @@ protected String internalEstimateGas(ProgramResult reversibleExecutionResult) {
             estimatedGas += GasCost.STIPEND_CALL;
         }
 
+        // ensure not returning blockGasLimit+stipend
+        estimatedGas = Math.min(estimatedGas, gasEstimationCap);
```
[the remark this answers is not in the document]
Comment (illuque-iov) [reply rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java:188]: Yes, as I understand it, that or some bug maybe. Will log it.

**Review by Vovchyk — APPROVED** [review]

**Review by asoto-iov — APPROVED** [review]

**Comment by Vovchyk** [comment]
pipeline:run

**Pull request merged**